go/vt/vtgate/planbuilder/operators/info_schema_planning.go

```diff
@@ -0,0 +1,361 @@
+	"SCHEMA_PRIVILEGES":       {"TABLE_SCHEMA"},
+	"STATISTICS":              {"TABLE_SCHEMA"},
+	"TABLE_CONSTRAINTS":       {"TABLE_SCHEMA", "CONSTRAINT_SCHEMA"},
+	"TABLE_PRIVILEGES":        {"TABLE_SCHEMA"},
+	"TABLES":                  {"TABLE_SCHEMA"},
+	"TRIGGERS":                {"TRIGGER_SCHEMA", "EVENT_OBJECT_SCHEMA"},
+	"VIEW":                    {"TRIGGER_SCHEMA"},
+}
+
+var schemaColName80 = map[string][]string{
+	"CHECK_CONSTRAINTS":            {"CONSTRAINT_SCHEMA"},
+	"COLUMN_PRIVILEGES":            {"TABLE_SCHEMA"},
+	"COLUMN_STATISTICS":            {"SCHEMA_NAME"},
+	"COLUMNS":                      {"TABLE_SCHEMA"},
+	"COLUMNS_EXTENSIONS":           {"TABLE_SCHEMA"},
+	"EVENTS":                       {"EVENT_SCHEMA"},
+	"FILES":                        {"TABLE_SCHEMA"},
+	"KEY_COLUMN_USAGE":             {"CONSTRAINT_SCHEMA", "TABLE_SCHEMA", "REFERENCED_TABLE_SCHEMA"},
+	"PARAMETERS":                   {"SPECIFIC_SCHEMA"},
+	"PARTITIONS":                   {"TABLE_SCHEMA"},
+	"REFERENTIAL_CONSTRAINTS":      {"CONSTRAINT_SCHEMA", "UNIQUE_CONSTRAINT_SCHEMA"},
+	"ROLE_COLUMN_GRANTS":           {"TABLE_SCHEMA"},
+	"ROLE_ROUTINE_GRANTS":          {"SPECIFIC_SCHEMA", "ROUTINE_SCHEMA"},
+	"ROLE_TABLE_GRANTS":            {"TABLE_SCHEMA"},
+	"ROUTINES":                     {"ROUTINE_SCHEMA"},
+	"SCHEMA_PRIVILEGES":            {"TABLE_SCHEMA"},
+	"SCHEMATA":                     {"SCHEMA_NAME"},
+	"SCHEMATA_EXTENSIONS":          {"SCHEMA_NAME"},
+	"ST_GEOMETRY_COLUMNS":          {"TABLE_SCHEMA"},
+	"STATISTICS":                   {"TABLE_SCHEMA"},
+	"TABLE_CONSTRAINTS":            {"TABLE_SCHEMA", "CONSTRAINT_SCHEMA"},
+	"TABLE_CONSTRAINTS_EXTENSIONS": {"CONSTRAINT_SCHEMA"},
+	"TABLE_PRIVILEGES":             {"TABLE_SCHEMA"},
+	"TABLES":                       {"TABLE_SCHEMA"},
+	"TABLES_EXTENSIONS":            {"TABLE_SCHEMA"},
+	"TRIGGERS":                     {"TRIGGER_SCHEMA", "EVENT_OBJECT_SCHEMA"},
+	"VIEW_ROUTINE_USAGE":           {"TABLE_SCHEMA", "SPECIFIC_SCHEMA"},
+	"VIEW_TABLE_USAGE":             {"TABLE_SCHEMA", "VIEW_SCHEMA"},
+	"VIEWS":                        {"TABLE_SCHEMA"},
+}
+
+func shouldRewrite(e sqlparser.Expr) bool {
+	switch node := e.(type) {
+	case *sqlparser.FuncExpr:
+		// we should not rewrite database() calls against information_schema
+		return !(node.Name.EqualString("database") || node.Name.EqualString("schema"))
+	}
+	return true
+}
+
+func isTableSchemaOrName(e sqlparser.Expr) (col *sqlparser.ColName, isTableSchema bool, isTableName bool) {
+	col, ok := e.(*sqlparser.ColName)
+	if !ok {
+		return nil, false, false
+	}
+	return col, isDbNameCol(col), isTableNameCol(col)
+}
+
+func isDbNameCol(col *sqlparser.ColName) bool {
+	return col.Name.EqualString("table_schema") || col.Name.EqualString("constraint_schema") || col.Name.EqualString("schema_name") || col.Name.EqualString("routine_schema")
+}
+
+func isTableNameCol(col *sqlparser.ColName) bool {
+	return col.Name.EqualString("table_name")
+}
```

Comment:
There are 2 more implementations of it in the `planbuilder` package, which have different definitions. We should merge them into one.